interp/interp.go

```diff
@@ -614,16 +693,10 @@ func (interp *Interpreter) REPL(in io.Reader, out io.Writer) {
 		prompt(v)
 	}
 	cancel() // Do not defer, as cancel func may change over time.
+	return v, err
```

Comment:
Maybe if err is nil at this point, we return s.Err() instead? But we have to see how that works out once this PR is rebased on top of master/PR 823.

Reply:
Right.